File: Python/underautomation/universal_robots/ssh/tools/shell.py
import typing
from underautomation.universal_robots.ssh.tools.common.exception_event_args import ExceptionEventArgs
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.Ssh.Tools import Shell as shell

class Shell:
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = shell()
		else:
			self._instance = _internal
	def starting(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.Starting+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def started(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.Started+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def stopping(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.Stopping+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def stopped(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.Stopped+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def error_occurred(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.ErrorOccurred+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def start(self) -> None:
		self._instance.Start()
	def stop(self) -> None:
		self._instance.Stop()
	@property
	def is_started(self) -> bool:
		return self._instance.IsStarted

File: Ssh/Tools/Shell.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using System;
using Ssh.Tools.Common;

namespace Ssh.Tools {
	/// <summary>
	/// Represents instance of the SSH shell object
	/// </summary>
	public class Shell : IDisposable {

		/// <summary>
		/// Gets a value indicating whether this shell is started.
		/// </summary>
		/// <returns><code>true</code> if started is started; otherwise, <code>false</code>.</returns>
		public bool IsStarted { get; }

		/// <summary>
		/// Occurs when shell is starting.
		/// </summary>
		public event EventHandler<EventArgs> Starting;

		/// <summary>
		/// Occurs when shell is started.
		/// </summary>
		public event EventHandler<EventArgs> Started;

		/// <summary>
		/// Occurs when shell is stopping.
		/// </summary>
		public event EventHandler<EventArgs> Stopping;

		/// <summary>
		/// Occurs when shell is stopped.
		/// </summary>
		public event EventHandler<EventArgs> Stopped;

		/// <summary>
		/// Occurs when an error occurred.
		/// </summary>
		public event EventHandler<ExceptionEventArgs> ErrorOccurred;

		/// <summary>
		/// Starts this shell.
		/// </summary>
		public void Start()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Stops this shell.
		/// </summary>
		public void Stop()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
		/// </summary>
		public void Dispose()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Releases unmanaged and - optionally - managed resources
		/// </summary>
		/// <param name="disposing"><code>true</code> to release both managed and unmanaged resources; <code>false</code> to release only unmanaged resources.</param>
		protected virtual void Dispose(bool disposing)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}
	}
}
